### PR TITLE
Add release steps to pipeline

### PR DESCRIPTION
**Pull request by sjberman**
The following steps occur when a tag is created:
1. goreleaser builds and uploads nginx-meshctl binaries
2. goreleaser publishes the release
3. goreleaser uploads nginx-meshctl sbom to azure
4. helm charts are published to nginx helm repo

What a release will look like (this is missing a couple artifacts that I added after taking the screenshot):
Screen Shot 2023-01-04 at 8 47 17 AM

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-service-mesh/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -29,5 +24,23 @@ checksum:
...
+    [Release Notes](https://docs.nginx.com/nginx-service-mesh/releases/release-notes-{{.Version}}/)
+
 archives:
-  - id: nginx-service-mesh
+  - id: nginx-meshctl
+    builds: [nginx-meshctl]
+    name_template: nginx-meshctl_{{ .Version }}_{{ .Os }}_{{ .Arch }}
+    format_overrides:
+      - goos: windows
+        format: zip
```
Comment (jbyers19) [inline .goreleaser.yml:40]: Is this only saving the Windows binary?

Comment (jbyers19) [reply .goreleaser.yml:40]: Or is this only for the purpose of generating the SBOM?

Comment (sjberman) [reply .goreleaser.yml:40]: By default, binaries are saved as `tar.gz`. This makes it so the windows binary is saved as `zip`.

Comment (sjberman) [reply .goreleaser.yml:40]: I added a screenshot to the description of what the release will look like.

**Review by jbyers19 — APPROVED** [review]

**Review by lucacome — APPROVED** [review]

**Review by pammecrandall — APPROVED** [review]

**Pull request merged**